qt/widgets/common/test/ProjectSaveModelTest.h

```diff
@@ -267,6 +281,36 @@ class ProjectSaveModelTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(winInfo[1].type, "Matrix");
     TS_ASSERT_EQUALS(winInfo[1].icon_id, "matrix_xpm");
   }
+
+  void testNeedsSizeWarningIsFalseWithEmptyWorkspace() {
+    std::vector<std::string> wsNames{"ws1"};
+    ProjectSaveModel model({});
+    TS_ASSERT_EQUALS(model.needsSizeWarning(wsNames), false);
+  }
+
+  void testNeedsSizeWarningIsTrueWithLargeWorkspace() {
+    std::vector<MantidQt::API::IProjectSerialisable *> windows;
+    NiceMock<MockProjectSaveModel> model(windows);
+    // std::unique_ptr<MockProjectSaveModel> model =
+    // std::make_unique<NiceMock<MockProjectSaveModel>>(windows);
+
+    std::vector<std::string> wsNames{"ws1", "ws2"};
+    ON_CALL(model, getProjectSize(wsNames)).WillByDefault(Return(107374182411));
+
+    TS_ASSERT(model.needsSizeWarning(wsNames));
+  }
+
+  void testGetProjectSizeReturnsCorrectAnswer() {
+    ProjectSaveModel model({});
+    const auto workspaces = model.getWorkspaces();
+    size_t assumedSize = 0;
+
+    for (const auto &ws : workspaces) {
```

Comment:
To make this test neater, you could create a function in an anonymous namespace to get the `assumedSize` from the `workspaces`.